### PR TITLE
deprecate sparse matmul op in CNN models

### PR DESCRIPTION
**Pull request by kamalrajkannan78**
## Summary

- This PR solves  [#641](https://github.com/tenstorrent/tt-forge-fe/issues/641) and adds a sanity test for upsample op in advance([#667](https://github.com/tenstorrent/tt-forge-fe/issues/667))
- Sparse matmul appeared in inceptionv4 due to [avgpool_2d](https://github.com/huggingface/pytorch-image-models/blob/9f5c279bad296447a0d896ac564846da677f93b2/timm/models/inception_v4.py#L84).So changes in pooling.py script are related only to sparse matmul deprecation in Inceptionv4 variants
- Due to [interpolation](https://github.com/qubvel-org/segmentation_models.pytorch/blob/d538393c645163a7f9dcda563320a417303fdf06/segmentation_models_pytorch/decoders/unet/decoder.py#L38) in decoder part, [resize2d decomposition](https://github.com/tenstorrent/tt-forge-fe/blob/5f06895685df5763bccef4d4dca133d773bcb9cc/forge/forge/op/eval/forge/resize.py#L196) in post initial graph pass leads to sparse matmul op occurence in unet. Changes in resize.py are related sparse matmul deprecation for models listed [here](https://github.com/tenstorrent/tt-forge-fe/issues/667#issue-2649610778)

## logs

- [interpolate_before_fix.log](https://github.com/user-attachments/files/17712316/nov12_interpolate_before_fix.log)
- [interpolate_after_fix.log](https://github.com/user-attachments/files/17712315/nov12_interpolate_after_fix.log)
- [inceptionv4_timm_before_fix.log](https://github.com/user-attachments/files/17707451/nov11_z_inceptionv4_timm_d1.log)
- [inceptionv4_timm_after_fix.log](https://github.com/user-attachments/files/17707450/nov11_z_inceptionv4_timm_full_model_after_fix.log)
- [inceptionv4_osmr_before_fix.log](https://github.com/user-attachments/files/17707475/nov11_z_inceptionv4_osmr_full_model_before_fix.log)
- [inceptionv4_osmr_after_fix.log](https://github.com/user-attachments/files/17707478/nov11_z_inceptionv4_osmr_full_model_after_fix.log)

### CODE REVIEW

File: forge/forge/op/eval/forge/pooling.py [hunk 1]
```diff
@@ -570,13 +574,17 @@ def decompose(type, attr, dc, inputs):
             # TODO: This sparse matmul can definitely be fused the same way the sparse mm of convtransposed2d was fused
```
Comment (nvukobratTT) [inline forge/forge/op/eval/forge/pooling.py:574]: Let's also remove this comment. Seems redundant right now

File: forge/test/mlir/test_ops.py [hunk 2]
```diff
@@ -14,6 +14,36 @@
...
+@pytest.mark.parametrize(
+    "shape, mode",
+    [
+        ((1, 2048, 7, 7), "nearest"),
+        ((1, 2048, 7, 7), "bilinear"),
+    ],
+)
+@pytest.mark.xfail(reason="Found Unsupported operations while lowering from TTForge to TTIR in forward graph")
+@pytest.mark.push
+def test_interpolate(shape, mode):
```
Comment (nvukobratTT) [inline forge/test/mlir/test_ops.py:26]: Do you plan to add upsample and downsample test cases in the following issues? 

Let's tr to push those tests and changes in next few PRs to get these model compiling through FFE :))

Comment (kamalrajkannan78) [reply forge/test/mlir/test_ops.py:26]: - [torch.nn.functional.upsample](https://pytorch.org/docs/stable/generated/torch.nn.functional.upsample.html) is equivalent to torch.nn.functional.interpolate. so above test is for upsample op
- None of models didn't need downsample op till now. so code related to it are left untouched and test is not added

Comment (nvukobratTT) [reply forge/test/mlir/test_ops.py:26]: Great! Thanks for confirmation Kamal :))

**Review by nvukobratTT — APPROVED** [review]

**Comment by nvukobratTT** [comment]
Approving PR, let's just remove mentioned comment :))

**Pull request merged**